### PR TITLE
refactor ApkDecoder - streamline decode procedure

### PR DESCRIPTION
**Pull request by sv99**
Add common methods decodeManifest and decodeResources.
Remove params wрich duplicates fields

### CODE REVIEW

File: brut.apktool/apktool-lib/src/main/java/brut/androlib/ApkDecoder.java [hunk 1]
```diff
@@ -53,7 +53,7 @@ public class ApkDecoder {
     private final static String SMALI_DIRNAME = "smali";
     private final static String UNK_DIRNAME = "unknown";
     private final static String[] APK_RESOURCES_FILENAMES = new String[] {
-        "resources.arsc", "AndroidManifest.xml", "res", "r", "R" };
+        "resources.arsc", "res", "r", "R" };
```
Comment (iBotPeaches) [inline brut.apktool/apktool-lib/src/main/java/brut/androlib/ApkDecoder.java:56]: Any background on removal of `AndroidManifest.xml` from this? In my vague memory we needed to duplicate it here so in a raw build (pure copy of compiled assets) we needed to include the Manifest.

Comment (sv99) [reply brut.apktool/apktool-lib/src/main/java/brut/androlib/ApkDecoder.java:56]: This private field used only in the decoder. In the builder has different list - don't touched

Comment (sv99) [reply brut.apktool/apktool-lib/src/main/java/brut/androlib/ApkDecoder.java:56]: I think need make class ApkArchive and method analyze():
a.  fill full entry list of the Apk
b. split this list to the category
  1. AndroidManifest.xml
  2. META-INF/ directory - with services
  3. resources.asrc
  4. resource files - liked from resources,asrc
  5. uncompressed files
 
  7. other or unknown files

Comment (iBotPeaches) [reply brut.apktool/apktool-lib/src/main/java/brut/androlib/ApkDecoder.java:56]: To work the features needed for `--analysis-mode` ?

Comment (sv99) [reply brut.apktool/apktool-lib/src/main/java/brut/androlib/ApkDecoder.java:56]: Yes, but we have use this splitted list for copy items. Don't need NO_COMPRESS_PATTERN - this information has in the apk.

**Review by iBotPeaches — APPROVED** [review]

**Comment by iBotPeaches** [comment]
thanks!

**Pull request merged**